docusaurus/docs/how-to-guides/app-plugins/add-backend-component.md

```diff
@@ -0,0 +1,145 @@
+
+## Access App settings
+
+Settings are part of the `AppInstanceSettings` struct. They are passed to the app plugin constructor as the second argument.
+
+```go title="src/app.go"
+func NewApp(ctx context.Context, settings backend.AppInstanceSettings) (instancemgmt.Instance, error) {
+  jsonData := settings.JSONData // json.RawMessage
+  secureJsonData := settings.DecryptedSecureJSONData // map[string]string
+}
+```
+
+You can also get the settings from a request `Context`
+
+```go title="src/resources.go"
+func (a *App) handleMyRequest(w http.ResponseWriter, req *http.Request) {
+  pluginConfig := backend.PluginConfigFromContext(req.Context())
+  jsonData := pluginConfig.AppInstanceSettings.JSONData // json.RawMessage
+}
+```
+
+## Add a custom endpoint to your app plugin
+
+### ServeMux (recommended)
+
+Your scaffoled app plugin already has a default CallResource that uses [ServeMux](https://pkg.go.dev/net/http#ServeMux).
+
+You can add custom endpoints to your app plugin in the `resources.go` file. E.g.:
+
+```go title="resources.go"
+// this function already exists in the scaffolded app plugin
+func (a *App) registerRoutes(mux *http.ServeMux) {
+	mux.HandleFunc("/myCustomEndpoint", a.handleMyCustomEndpoint)
+}
+
+func (a *App) handleMyCustomEndpoint(w http.ResponseWriter, r *http.Request) {
+  // handle the request
+  // e.g. call a third-party API
+  w.Write([]byte("my custom response"))
+  w.WriteHeader(http.StatusOK)
+}
+```
+
+### CallResource
+
+You can also add custom endpoints to your app plugin by adding a CallResource handler to your backend component directly. You will have to implement the logic to handle multiple requests.
+
+```go title="app.go"
+func (a *App) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	switch req.Path {
+	case "myCustomEndpoint":
+		sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusOK,
+			Body:   []byte("my custom response"),
+		})
+	default:
+		return sender.Send(&backend.CallResourceResponse{
+			Status: http.StatusNotFound,
+		})
+	}
+	return nil
+}
+```
+
+You can also see the data sources [documentation on resource handler](../data-source-plugins/add-resource-handler.md) which you can also apply to your app plugin.
+
+### Calling your custom endpoint from frontend code
+
+To call your custom endpoint from frontend code, you can use the `fetch` function from `getBackendSrv`. E.g.:
+
+```ts
+import { getBackendSrv } from '@grafana/runtime';
+import { lastValueFrom } from 'rxjs';
+
+function getMyCustomEndpoint() {
+  const response = await getBackendSrv().fetch({
+    // replace ${PLUGIN_ID} with your plugin id
+    url: '/api/plugins/${PLUGIN_ID}/myCustomEndpoint',
+  });
+  return await lastValueFrom(response);
+}
+```
+
```

Comment:
Also maybe worth it to point to our example to see the fully working code: https://github.com/grafana/grafana-plugin-examples/tree/main/examples/app-with-backend